include/miral/miral/config_file.h

```diff
@@ -56,6 +57,8 @@ class ConfigFile
     ConfigFile(MirRunner& runner, std::filesystem::path file, Mode mode, Loader load_config);
     ~ConfigFile();
 
+    void operator()(mir::Server&);
+    void load_and_watch();
```

Comment:
Unfortunately, changing the way this class is used breaks the API. We should try to find an alternative.

I think the best way to achieve that would be to use `MirRunner::add_start_callback()` in the constructor to `self->load_and_watch()`.

In any case, there should be no need for `ConfigFile::load_and_watch()` - that's an implementation detail that users need not be exposed to.